tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -104,10 +106,9 @@ std::vector<at::Tensor> reference_mlp(
     at::Tensor w1,
     at::Tensor b1,
     at::ScalarType at_dtype) {
-  auto linear0 = at::matmul(x, w0).to(at::kFloat) + b0.to(at::kFloat);
-  auto gelu = at::gelu(linear0, "tanh");
-  auto linear1 =
-      at::matmul(gelu.to(at_dtype), w1).to(at::kFloat) + b1.to(at::kFloat);
+  auto linear0 = at::matmul(x, w0).to(at::kFloat) + b0;
+  auto gelu = at::gelu(linear0, "tanh").to(at_dtype);
```

Comment:
This is necessary for `at::allclose` to pass because `mlp` generates the output of `at_dtype`.